### ISSUE TITLE
Running CrystalReportsNinja in Php

### ISSUE THREAD

**Issue by abnertole** [rainforestnet/CrystalReportsNinja#2]
Hi, 

Thanks for sharing the tool. I have tried it and encountered the error below.

When I run in windows CMD, the crystal report is generated. But when I execute via Php exec(), I am getting the error below. Anything I missed out?

Exception: Unknown Database Connector Error Error in File OPRJ 9496_9704_{1CA7AD4F-AFE6-4FA9-8840-B945674BF453}.rpt: Unknown Database Connector Error Inner Exception: System.Runtime.InteropServices.COMException (0x800002C2): Unknown Database Connector Error Error in File OPRJ 9496_9704_{1CA7AD4F-AFE6-4FA9-8840-B945674BF453}.rpt: Unknown Database Connector Error at CrystalDecisions.ReportAppServer.Controllers.ReportSourceClass.Export(ExportOptions pExportOptions, RequestContext pRequestContext) at CrystalDecisions.ReportSource.EromReportSourceBase.ExportToStream(ExportRequestContext reqContext) 

Thanks.

Abner

**Comment by rainforestnet** [comment]
@abnertole I have never tried to use php to run a .net executable before. I am not sure of that.
I did have a asp.net project that call this and it works.

**Comment by fmartiner** [comment]
@abnertole 
Set the correct ORACLE_HOME and TNS_ADMIN path in environment variable before launch cmd of CrystalReportsNinja.exe from PHP
Example create a .bat file with:

SET ORACLE_HOME=C:\...
SET TNS_ADMIN=C:\...
CrystalReportsNinja.exe -F ...

and exec(file.bat)

This solve for me.

**Comment by mikecrf121** [comment]
@rainforestnet curious, how did you run this from another .net application??  did your app open a command line and then pass strings to the console, im considering doing something simular ??

**Comment by mharned73** [comment]
Opened from a exe statement is SQL Server Stored Procedure….  The command is passed as a string

Sent from Mail<https://go.microsoft.com/fwlink/?LinkId=550986> for Windows 10

________________________________

**Comment by rainforestnet** [comment]
@mikecrf121 check this out
https://stackoverflow.com/questions/9679375/run-an-exe-from-c-sharp-code

**Comment by mikecrf121** [comment]
appreciate it thank you!  what im trying to do is receive a json object, parse it with a node app and feed it to CRN.  very useful library good job!

**Comment by mikecrf121** [comment]
@rainforestnet hey quick question, just now running into this issue, is this able to work with generating reports that contain sub reports?

Thanks!

**Comment by mharned73** [comment]
Yes.... from the point of executing runtime it is just crystal reports

Get Outlook for Android<https://aka.ms/ghei36>

________________________________

**Comment by mikecrf121** [comment]
What if the report references another .rpt file as the sub report?  does it have to be located within the same directory or how does that work?

**Comment by mikecrf121** [comment]
@mharned73 @rainforestnet  Hi, sorry to bother you but you both know the most about this.  So two things, having trouble generating reports that contain sub reports, and also I am only able to get this to work(Ninja) with Cr runtime 20 , when the newest one is 25, any ideas on either of these or any help would be much appreciated!!!!!

**Comment by aj1432** [comment]
hello team Please Explain Step by Step Please
I do not understand Command.

please check attachment, I got error
![image](https://user-images.githubusercontent.com/38251097/154800366-68c82388-3ae1-43da-b373-d26320894d27.png)

**Comment by mikecrf121** [comment]
> 

@aj1432
What do you mean you don't understand command? Like you can't run CRN at all or you don't understand the parameters?  I haven't worked with CRN in years... but this is an amazing amazing program, especially to save money if your facing the same problem my company did at the time.  You need to compile it first, what are you using to do that?? If you dont know what im talking about or how to even do that to make code changes if needed then id suggest hiring someone who knows what they are doing, like me haha.  Your also trying to run ,from your screen shot the first time with no required param "-F"... If you want I can show you on a one on one with a zoom call?  I may charge you to refresh myself though.  What are you trying to use crystal report ninja to do?  I remember when I did it for the company I was working for a few years ago, we built a custom angular app, with a node backend that executed CRN from the node app and returned the generated crystal reports to AWS S3, and we utilized sqs, sns, lambdas, etc... but anyways we built this app to avoid ridiculous licensing costs associated with generating crystal reports from some program, it was like a per user cost that our company was trying to save money by avoiding the high costs of multiple licenses.  I'm also able to build out an app for you if your hiring or interested in contracting me. Can be part time or for a few hours, depending on your needs.  Thanks and good luck.
